src/tests/fuzz/fuzz_tcpsplit.c

```diff
@@ -0,0 +1,463 @@
+        - tftp\n\
+        - ikev2\n\
+        - krb5\n\
+        - snmp\n\
+        - rfb\n\
+        - sip\n\
+        - dhcp:\n\
+            enabled: yes\n\
+            extended: yes\n\
+        - ssh\n\
+        - flow\n\
+        - netflow\n\
+        - metadata\n\
+  - http-log:\n\
+      enabled: yes\n\
+      filename: /dev/null\n\
+      extended: yes\n\
+  - tls-log:\n\
+      enabled: yes\n\
+      filename: /dev/null\n\
+      extended: yes\n\
+app-layer:\n\
+  protocols:\n\
+    rdp:\n\
+      enabled: yes\n\
+    modbus:\n\
+      enabled: yes\n\
+      detection-ports:\n\
+        dp: 502\n\
+    dnp3:\n\
+      enabled: yes\n\
+      detection-ports:\n\
+        dp: 20000\n\
+    enip:\n\
+      enabled: yes\n\
+      detection-ports:\n\
+        dp: 44818\n\
+        sp: 44818\n\
+    sip:\n\
+      enabled: yes\n\
+";
+
+TcpReassemblyThreadCtx *ra_ctx = NULL;
+uint64_t forceLayer = 0;
+
+static Packet *TestHelperBuildPacketAndFlow(const char *src, const char *dst,
+                                     uint16_t sport, uint16_t dport)
+{
+    struct in_addr in;
+
+    Packet *p = PacketGetFromAlloc();
+    if (unlikely(p == NULL))
+        return NULL;
+
+    struct timeval tv;
+    TimeGet(&tv);
+    COPY_TIMESTAMP(&tv, &p->ts);
+
+    p->src.family = AF_INET;
+    p->dst.family = AF_INET;
+    p->payload = NULL;
+    p->payload_len = 0;
+    p->proto = IPPROTO_TCP;
+
+    if (inet_pton(AF_INET, src, &in) != 1)
+        goto error;
+    p->src.addr_data32[0] = in.s_addr;
+    p->sp = sport;
+
+    if (inet_pton(AF_INET, dst, &in) != 1)
+        goto error;
+    p->dst.addr_data32[0] = in.s_addr;
+    p->dp = dport;
+
+    p->ip4h = (IPV4Hdr *)GET_PKT_DATA(p);
+    if (p->ip4h == NULL)
+        goto error;
+
+    p->ip4h->s_ip_src.s_addr = p->src.addr_data32[0];
+    p->ip4h->s_ip_dst.s_addr = p->dst.addr_data32[0];
+    p->ip4h->ip_proto = IPPROTO_TCP;
+    p->ip4h->ip_verhl = sizeof(IPV4Hdr);
+
+    int hdr_offset = sizeof(IPV4Hdr);
+    p->tcph = (TCPHdr *)(GET_PKT_DATA(p) + sizeof(IPV4Hdr));
+    if (p->tcph == NULL)
+        goto error;
+    p->tcph->th_sport = htons(sport);
+    p->tcph->th_dport = htons(dport);
+    hdr_offset += sizeof(TCPHdr);
+
+    SET_PKT_LEN(p, hdr_offset);
+    p->payload = GET_PKT_DATA(p)+hdr_offset;
+
+    //no UTHBuildFlow to have storage
+    Flow *f = FlowAlloc();
+    if (f == NULL) {
+        goto error;
+    }
+    f->flags |= FLOW_IPV4;
+    f->src.addr_data32[0] = p->src.addr_data32[0];
+    f->dst.addr_data32[0] = p->dst.addr_data32[0];
+    f->sp = sport;
+    f->dp = dport;
+    f->proto = IPPROTO_TCP;
+    f->protomap = FlowGetProtoMapping(IPPROTO_TCP);
+    p->flow = f;
+    p->flags |= PKT_HAS_FLOW;
+    p->flags |= PKT_STREAM_EST;
+
+    return p;
+
+error:
+    SCFree(p);
+    return NULL;
+}
+
+static int TestHelperAddSegmentWithPayload(ThreadVars *tv, TcpReassemblyThreadCtx *ra_ctx, TcpStream *stream, uint32_t seq, const uint8_t *payload, uint16_t len, Packet *p)
+{
```

Comment:
inspired by StreamTcpUTAddSegmentWithPayload

The whole thing is inspired by tests such as `DetectSshVersionTestDetect01` from ssh rust branch